mantidproject/mantid
# Add class to get workspace history

**Description of work.**

This updates the error reporter so that if the user decided to share information the recovery folder is compressed and sent to the server along with the error report. 

**Report to:** keith 

**To test:**
* Select erroreports.mantidproject.org as your url
* Copy the file below into your mantidBuild/bin directory and run it with `python server_test.py`
* All the tests should pass but you should get some errors about failing to upload the recovery files.
[server_test.py.zip](https://github.com/mantidproject/mantid/files/2392633/server_test.py.zip)






Fixes #23395 . 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
